src/configs/axios.config.ts

```diff
@@ -4,24 +4,33 @@ import axios from 'axios';
 
 const URL = import.meta.env.VITE_API_URL;
 
-export const axiosInstance = axios.create({
+export const api = axios.create({
   baseURL: URL,
   withCredentials: true,
   headers: {
     'Content-Type': 'application/json',
   },
 });
 
-// TODO: further work on interceptor required
-axiosInstance.interceptors.response.use(
+api.interceptors.response.use(
   (response) => response,
   async (error) => {
-    if (error.response && error.response?.status === 401 && error.message !== 'Invalid refresh token') {
+    const origReq = error.config;
+
+    let retryCnt = Number(origReq.headers['X-Refresh-Token-Retry-Count']);
```

Comment:
Add a type check for headers. It may cause an error if it doesn't exist.

Reply:
no, it doesn't, I checked